### PR TITLE
The gnss_d300 driver based on dora

### PR DESCRIPTION
**Pull request by Cyanglll**
# Hardware Name or Vendor
GNSS_D300
## Description

Position positioning: By receiving satellite signals, GNSS_D300 can locate its own position (longitude, latitude and altitude).This driver is for the USB interface of GNSS hardware.
Completed the release of the NavSatFix, QuaternionStamped, and TwistStamped speed parts corresponding to ros2.

### Installation
Serial port connected to computer USB port.
The default value in nmea_getSentence_driver_dora.py is changed according to the situation after connecting the serial port:
serial_port = '/dev/ttyUSB1'
serial_baud = 115200

### Development Guidelines

sudo chmod 777 /dev/ttyUSB*
RUST_LOG=debug dora start gnssD300_nmea_driver_dora.yaml  --attach --hot-reload

### Output
Completed the release of the NavSatFix, QuaternionStamped, and TwistStamped speed parts corresponding to ros2.

- [x] Create a new folder for each new hardware under the right categorisation.
- [x] Add this README template into the folder as well as the source code.

### CODE REVIEW

**Comment by Cyanglll** [comment]
@haixuanTao Excuse me sir, can you please review this submission and give me some advice，thank you!

File: vendors/gnss/GNSS_D300/__pycache__/DoraNmeaDriver_utils.cpython-37.pyc
Comment (haixuanTao) [inline vendors/gnss/GNSS_D300/__pycache__/DoraNmeaDriver_utils.cpython-37.pyc:1]: Can we remove those `__pycache__` file please`

File: vendors/gnss/GNSS_D300/nmea_subscribeSentence_driver_dora.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            dora_original_input = "ddd".encode()
+            send_output(
+                "hh",
+                dora_original_input,
+                dora_input["metadata"],
+            )
+            return DoraStatus.CONTINUE
+
+        else:
+            return DoraStatus.CONTINUE
```
Comment (haixuanTao) [inline vendors/gnss/GNSS_D300/nmea_subscribeSentence_driver_dora.py]: Can we format file using [black](https://github.com/psf/black)
```
pip install black
black .
```
There is a newline missing in this file that is necessary according to PEP8.

You can add format-on-save black on most IDE normally.

File: vendors/gnss/GNSS_D300/README [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+The default value in nmea_getSentence_driver_dora.py is changed according to the situation after connecting the serial port:
+
+serial_port = '/dev/ttyUSB1'
+serial_baud = 115200
+
+2.1.nmea_publishSentence_driver_dora：Complete the judgment and analysis of the input NMEA sentence format, and the analysis is correct, and send the serialized class send_out to nmea_subscribeSentence_driver_dora.
+
+3.nmea_subscribeSentence_driver_dora：Discriminate the message type and output the parsed data.
+
```
Comment (haixuanTao) [inline vendors/gnss/GNSS_D300/README]: Why do we needs 3 operators?
Can't we just have one operator?

File: vendors/gnss/GNSS_D300/__pycache__/DoraNmeaDriver_utils.cpython-37.pyc
Comment (Cyanglll) [reply vendors/gnss/GNSS_D300/__pycache__/DoraNmeaDriver_utils.cpython-37.pyc:1]: yes, i will remove them.

File: vendors/gnss/GNSS_D300/nmea_subscribeSentence_driver_dora.py (hunk 1 above)
Comment (Cyanglll) [reply vendors/gnss/GNSS_D300/nmea_subscribeSentence_driver_dora.py]: ok,thank u!

File: vendors/gnss/GNSS_D300/README (hunk 2 above)
Comment (Cyanglll) [reply vendors/gnss/GNSS_D300/README]: "Okay, to make it more logical, I'll put the data sending code in one operator and the data receiving code in another operator."

Comment (ChenyangLv529) [reply vendors/gnss/GNSS_D300/README]: Excue me, I just resubmitted my edits and in the latest file：
I put the data sending code in the operator of "gnss_D300_driver_dora" and the data receiving code in another operator of "nmea_subscribeTopic". By the way, the custom classes and methods used in above two operators  are placed in "DoraNmeaDriver_utils.py" and "Nmea_utils.py".

Comment (Cyanglll) [reply vendors/gnss/GNSS_D300/README]: @haixuanTao Excue me, I just resubmitted my edits and in the latest file：
I put the data sending code in the operator of "gnss_D300_driver_dora" and the data receiving code in another operator of "nmea_subscribeTopic". By the way, the custom classes and methods used in above two operators are placed in "DoraNmeaDriver_utils.py" and "Nmea_utils.py".

**Review by bobdingAI — APPROVED** [review]

**Review by bobdingAI — APPROVED** [review]

**Pull request merged**